shell/browser/browser_win.cc

```diff
@@ -68,13 +68,14 @@ bool GetProtocolLaunchPath(gin::Arguments* args, std::wstring* exe) {
 
   // Read in optional args arg
   std::vector<std::wstring> launch_args;
-  if (args->GetNext(&launch_args) && !launch_args.empty())
-    *exe = base::UTF8ToWide(
-        base::StringPrintf("\"%ls\" \"%ls\" \"%%1\"", exe->c_str(),
-                           base::JoinString(launch_args, L"\"  \"").c_str()));
-  else
-    *exe =
-        base::UTF8ToWide(base::StringPrintf("\"%ls\" \"%%1\"", exe->c_str()));
+  if (args->GetNext(&launch_args) && !launch_args.empty()) {
+    std::wstring joined_args = base::JoinString(launch_args, L"\" \"");
+    *exe = base::StrCat(
+        {L"\"", exe->c_str(), L"\" \"", joined_args.c_str(), L"\" \"%1\""});
```

Comment:
_(nit)_ Looks like this is invoking `std::u16string StrCat(span<const StringPiece16>)`. If so, then adding the `.c_str()` here is a minor pessimization since it discards the original's string length info and forces the StringPiece16 constructor to re-calculate the same length.

```suggestion
        {L"\"", *exe, L"\" \"", joined_args, L"\" \"%1\""});
```

(I don't have a Windows dev box handy to confirm with though!)

Reply:
Checking on my windows machine, this is invoking `std::wstring StrCat(std::initializer_list<std::wstring_view> pieces)` from `//base/strings/strcat_win.h`. That said, your advice still holds and it looks like it works on my machine, so I'm going to make these changes. 👍

(Chromium does a lot of "poison checking" on `StrCat` on Windows... I don't fully understand, but it makes it hard to jump-to-definition of `base::StrCat` with clangd 😭.)